packages/frontend/app/components/ChangeLogModal.tsx

```diff
@@ -0,0 +1,75 @@
+import { Modal } from 'antd'
+import { useEffect, useState } from 'react'
+import { useUserInfo } from '../contexts/userContext'
+import MarkdownCustom from './Markdown'
+import { API } from '../api'
+
+interface ChangeLogModalProps {
+  isOpen: boolean
+  setIsOpen: (isOpen: boolean) => void
+}
+
+const ChangeLogModal: React.FC<ChangeLogModalProps> = ({
+  isOpen,
+  setIsOpen,
+}) => {
+  const { userInfo, setUserInfo } = useUserInfo()
+  const [currentChangeLog, setCurrentChangeLog] = useState<string>('')
+
+  useEffect(() => {
+    if (!userInfo?.readChangeLog) {
```

Comment:
I have an idear: I'm thinking the popup modal might be a little annoying, especially for new students or users that barely use the system (like they're just here to join the queue). So, I'm thinking instead to NOT show the modal if there's new changes, and instead show "(new changes!)" right beside the version number in the footer (which should be pretty easy to do).

Also you probably want a strict equality userInfo?.readChangeLog === False since otherwise it may run True even when the data is just loading

I can also make these changes later today